packages/flutter/lib/src/material/elevated_button.dart

```diff
@@ -283,53 +281,53 @@ class ElevatedButton extends ButtonStyleButton {
   /// Returns the [ElevatedButtonThemeData.style] of the closest
   /// [ElevatedButtonTheme] ancestor.
   @override
-  ButtonStyle themeStyleOf(BuildContext context) {
-    return ElevatedButtonTheme.of(context)?.style;
+  ButtonStyle? themeStyleOf(BuildContext context) {
+    return ElevatedButtonTheme.of(context).style;
   }
 }
 
 @immutable
-class _ElevatedButtonDefaultBackground extends MaterialStateProperty<Color> with Diagnosticable {
+class _ElevatedButtonDefaultBackground extends MaterialStateProperty<Color?> with Diagnosticable {
   _ElevatedButtonDefaultBackground(this.primary, this.onSurface);
 
-  final Color primary;
-  final Color onSurface;
+  final Color? primary;
+  final Color? onSurface;
 
   @override
-  Color resolve(Set<MaterialState> states) {
+  Color? resolve(Set<MaterialState> states) {
     if (states.contains(MaterialState.disabled))
       return onSurface?.withOpacity(0.12);
     return primary;
   }
 }
 
 @immutable
-class _ElevatedButtonDefaultForeground extends MaterialStateProperty<Color> with Diagnosticable {
+class _ElevatedButtonDefaultForeground extends MaterialStateProperty<Color?> with Diagnosticable {
   _ElevatedButtonDefaultForeground(this.onPrimary, this.onSurface);
 
-  final Color onPrimary;
-  final Color onSurface;
+  final Color? onPrimary;
+  final Color? onSurface;
 
   @override
-  Color resolve(Set<MaterialState> states) {
+  Color? resolve(Set<MaterialState> states) {
```

Comment:
cc @HansMuller is it ok for this to return null?

Reply:
Yes. If it returns null, the caller is responsible for looking up a default.